### PR TITLE
Feature/socket assoc

### PR DESCRIPTION
**Pull request by kdvalin**
This PR associates `SocketHandler` structs with a process, allowing easy identification of what messages came from what plugin.

This PR also includes integration tests for Plugin/SocketHandler interactions

### CODE REVIEW

File: src/core/mod.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
-use socket_handler::SocketHandler;
-
 use anyhow::Result;
 
+use crate::process_management::process_manager::ProcessManager;
+
 pub struct Core {
-    socket_hadler: SocketHandler
+    proc_manager: ProcessManager
```
Comment (jaredoconnell) [inline src/core/mod.rs:8]: Let's do something with this unused variable next PR.

File: src/process_management/process.rs [hunk 2]
```diff
@@ -1,43 +1,99 @@
...
 use anyhow::Result;
+use serde::Serialize;
+use tokio::{task::JoinHandle, sync::{ Mutex, mpsc::{self, Receiver, Sender}}, time::timeout};
+
 #[derive(Debug)]
 pub struct Process {
     child: Child,
+    process_path: PathBuf,
+    core_read_thread: JoinHandle<()>,
+    socket: Arc<Mutex<SocketHandler>>,
```
Comment (jaredoconnell) [inline src/process_management/process.rs:23]: Do you think we'll be able to lock it separately in the future to allow concurrent read and writes? That should work with IPC.

File: tests/process_socket_int_test.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    #[test_log::test(tokio::test)]
+    async fn test_send_plugin_inst(#[case] ins_type: PluginInstructionType) {
+        let name = format!("polychat_process_send_plugin_inst_{}", ins_type);
+        let mut proc = assert_ok!(Process::new(TEST_PROGRAM, create_socket_server(&name)));
+        let mut comms = create_socket_client(&name).await;
+
+        let core_payload = SerializableCoreInstr {
+            instruction_type: CoreInstructionType::Init,
+            payload: create_core_payload()
+        };
```
Comment (jaredoconnell) [inline tests/process_socket_int_test.rs]: Can we add some comments about the test? In this case it appears to me to be starting a random process, and separately verifying that the socket works.

**Review by jaredoconnell — CHANGES_REQUESTED** [review]
I think all we need is a few more comments.

**Review by jaredoconnell — APPROVED** [review]
I added comments to the integration tests that describe my understanding of the code.
If those comments are good, feel free to merge the PR.

File: src/process_management/process.rs (hunk 2 above)
Comment (kdvalin) [reply src/process_management/process.rs:23]: Would require some work on SocketHandler..  Exactly how much work that would entail is unknown, let's open an issue for this and then revisit in a different PR.

**Comment by kdvalin** [comment]
Commets LGTM

File: src/process_management/process.rs (hunk 2 above)
Comment (jaredoconnell) [reply src/process_management/process.rs:23]: Issue created.

**Pull request merged**